gems/pending/appliance_console/database_replication.rb

```diff
@@ -0,0 +1,93 @@
+require 'appliance_console/logging'
+require 'appliance_console/prompts'
+require 'English'
+
+module ApplianceConsole
+  class DatabaseReplication < DatabaseConfiguration
+    include ApplianceConsole::Logging
+
+    REPMGR_CONFIG = '/etc/repmgr.conf'.freeze
+
+    attr_accessor :cluster_name, :node_number
+
+    def initialize(hash = {})
+      super
+    end
+
+    def ask_for_cluster_name
+      self.cluster_name = just_ask("cluster name", cluster_name)
+    end
+
+    def ask_for_node_number
+      self.node_number = ask_for_integer("node number")
+    end
+
+    def confirm
+      clear_screen
+      say(<<-EOL)
+Replication Server Configuration
+
+        Cluster Name: #{cluster_name}
+        Node Number:  #{node_number}
+        Host:         #{host}
+        Username      #{username}
+        Database      #{database}
+        Password      #{password}
+        EOL
+
+      agree("Apply this Replication Master Configuration? (Y/N): ")
+    end
+
+    def already_configured?
+      if File.exist?(REPMGR_CONFIG)
+        say("Warning: File #{REPMGR_CONFIG} exists. Replication is already configured")
+        Logging.logger.info("Warning: File #{REPMGR_CONFIG} exists. Replication is already configured")
+        return agree("Continue with configuration? (Y/N): ")
+      end
+      true
+    end
+
+    def create_config_file(node)
+      File.open(DatabaseReplication::REPMGR_CONFIG, "w") do |f|
+        f.puts "cluster=#{cluster_name}"
+        f.puts "node=#{node_number}"
+        f.puts "node_name=#{node}"
+        f.puts "conninfo='host=#{node} user=#{username} dbname=#{database}'"
+        f.puts "use_replication_slots=1"
+        f.puts "pg_basebackup_options='--xlog-method=stream'"
+      end
+      true
+    end
+
+    def run_as_postgres
+      pid = Process.fork do
+        Process::UID.change_privilege(Process::UID.from_name("postgres"))
+        yield if block_given?
+      end
+
+      Process.wait(pid)
+      exit_status = $CHILD_STATUS.exitstatus
+      return true if exit_status == 0
+      false
+    end
+
+    def configure_replication_server(cmd, params = {})
+      run_as_postgres do
+        begin
+          res = AwesomeSpawn.run!(cmd, :params => params, :env => {"PGPASSWORD" => password})
+          say res.output
+        rescue AwesomeSpawn::CommandResultError => e
+          say e.result.output
+          say e.result.error
+          say ""
+          say("Failed to configure replication server")
+          exit 1
+        rescue => e
+          say("Failed to configure replication server #{e.message}")
+          exit 1
+        end
+        exit 0
```

Comment:
I don't think this is needed.

Also it's a bit odd that the caller of `run_as_postgres` needs to know that `run_as_postgres` is forking so the bock should `exit` instead of running normally.

I think the exception handling should probably be moved into `run_as_postgres` so that the exit status is also handled by the only piece of code that knows it is forking a new process.